test_requirements.txt

```diff
@@ -1,4 +1,5 @@
 cython==0.29.13
+hypothesis==4.35.0
```

Comment:
The version should not be pinned. Hypothesis releases very frequently, and pinning means that dependencybot will issue a PR for each new version. Let's just assume the deprecations or API changes will be less frequent than the releases.

Reply:
In fact there have been three since this PR was opened!

Deprecations or API changes *that affect Numpy's tests* should be pretty rare, but do specify a minimum version since we've been adding shape-related strategies lately and have some more in the pipeline :smile:

Reply:
@kitchoi could you change this to `>=` ? Let's see how dependencybot handles that...

Reply:
Is this actually necessary? It looks like #14411 put dependabot on a weekly schedule, regardless of how many times Hypothesis updates.

Reply:
So by including hypothesis we pretty much guarantee we will get at least one weekly PR from dependabot (assuming the configuration does what it claims) to update dependencies? That sounds like a significant maintenance burden. Maybe we should instead document its use during development, from the [comment elsewhere](https://github.com/numpy/numpy/pull/14440#issuecomment-532077648) in the PR and not make it a hard dependency for CI.

Reply:
Hypothesis development is fairly bursty - for the last few months we've had a release in less than half of all weeks.  That said, if a weekly PR for dependency updates is too often, you could [set dependabot to monthly updates](https://dependabot.com/docs/config-file/#update_schedule-required).

I'd also suggest investing some time in making dependency updates easier; for Hypothesis it's rare that I would do more than click approve and click merge.  For gnarly-looking interactions, I usually just close the PR and get upstream's fixes a week later - getting a proposed update every week doesn't mean you have to accept them all!